library/Vanilla/UploadedFile.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GNU GPLv2
+ */
+
+namespace Vanilla;
+
+use Gdn;
```

Comment:
You can't have a dependency on Gdn in the Vanilla namespace.

Receive the container or Gdn_Upload in the constructor.